app/models/compute_resource.rb

```diff
@@ -52,25 +41,44 @@ class ComputeResource < ActiveRecord::Base
     end
   }
 
-  def self.register_provider(provider)
-    name = provider.name.split('::').last
-    return if supported_providers.values.include?(provider) || supported_providers.keys.include?(name)
-    supported_providers[name] = provider.name
+  def self.supported_providers
+    {
+      'Libvirt'   => 'Foreman::Model::Libvirt',
+      'Ovirt'     => 'Foreman::Model::Ovirt',
+      'EC2'       => 'Foreman::Model::EC2',
+      'Vmware'    => 'Foreman::Model::Vmware',
+      'Openstack' => 'Foreman::Model::Openstack',
+      'Rackspace' => 'Foreman::Model::Rackspace',
+      'GCE'       => 'Foreman::Model::GCE',
+    }
+  end
+
+  def self.registered_providers
+    Foreman::Plugin.all.map(&:compute_resources).inject({}) do |prov_hash,providers|
```

Comment:
One nitpick `prov_hash,providers` space after comma would be nice: on the other hand we have the cop disabled right now and would be fixed once enabled one day, so feel free to ignore this :)